contracts/state_transaction.go

```diff
@@ -0,0 +1,42 @@
+package contracts
+
+import (
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type StateSyncProof struct {
+	Proof     []types.Hash
+	StateSync *contractsapi.StateSyncedEvent
+}
+
+// EncodeAbi contains logic for encoding given ABI data
+func (ssp *StateSyncProof) EncodeAbi() ([]byte, error) {
+	execute := contractsapi.ExecuteFunction{
+		Proof: ssp.Proof,
+		Obj:   (*contractsapi.StateSync)(ssp.StateSync),
+	}
+
+	return execute.EncodeAbi()
+}
+
+// DecodeAbi contains logic for decoding given ABI data
+func (ssp *StateSyncProof) DecodeAbi(txData []byte) error {
+	execute := &contractsapi.ExecuteFunction{}
+
+	if err := execute.DecodeAbi(txData); err != nil {
+		return err
+	}
+
+	*ssp = StateSyncProof{
```

Comment:
just a suggestion (maybe not of importance at all :) ), maybe it is easier to understand what is happening if you write:
```
ssp.Proof =    execute.Proof,
ssp.StateSync = (*contractsapi.StateSyncedEvent)(execute.Obj)
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1166/commits/10ea6a07f13637573f8862bd081fdcccf62f7836